cpp/src/arrow/flight/types.cc

```diff
@@ -118,6 +118,19 @@ std::string FlightDescriptor::ToString() const {
   return ss.str();
 }
 
+Status SchemaResult::GetSchema(ipc::DictionaryMemo* dictionary_memo,
+                               std::shared_ptr<Schema>* out) const {
+  if (reconstructed_schema_) {
+    *out = schema_;
+    return Status::OK();
+  }
+  io::BufferReader schema_reader(raw_schema_);
```

Comment:
is there any reason to do this work lazily?  Can parsing be done in a static factory method instead (and make the constructed private?

Reply:
Ahhh, missed that comment. Will fix asap. I was following the pattern from getflightinfo but I agree that it's not needed here

Reply:
Ahhh, missed that comment. Will fix asap. I was following the pattern from getflightinfo but I agree that it's not needed here

Reply:
Hey @emkornfield not sure what you mean here actually. GetSchema should be run lazily if it is to reuse an existing dictionary_memo. What were you thinking should be put in the static method and which constructor is private. Do you mean SchemaResult has a private constructor? That is fine by me but the schema would still require a dict memo to convert from a string.

Reply:
I prefer option 1 personally and have pushed a change implementing it. An example usage of the method: https://github.com/apache/arrow/pull/4980/files#diff-6515fea34c5e970d6d5a54c688652e54R535

Though this implies that the same method in FlightInfo should be changed https://github.com/apache/arrow/blob/master/cpp/src/arrow/flight/types.h#L344

I can:
1. do that here
2. open up a ticket
3. open up a new PR